src/matchOptimisticB.cc

```diff
@@ -0,0 +1,672 @@
+            throw LSST_EXCEPT(pexExcept::InvalidParameterError, "angleDiffFrom90 must be positive");
+        }
+        if (numPointsForShape <= 0) {
+            throw LSST_EXCEPT(pexExcept::InvalidParameterError, "numPointsForShape must be positive");
+        }
+        if (maxDeterminant <= 0) {
+            throw LSST_EXCEPT(pexExcept::InvalidParameterError, "maxDeterminant must be positive");
+        }
+    }
+
+    ProxyVector makeProxies(afwTable::SourceCatalog const & sourceCat) {
+        ProxyVector r;
+        r.reserve(sourceCat.size());
+        for (afwTable::SourceCatalog::const_iterator sourcePtr = sourceCat.begin();
+            sourcePtr != sourceCat.end(); ++sourcePtr) {
+            r.push_back(RecordProxy(sourcePtr, sourcePtr->getCentroid()));
+        }
+        return r;
+    }
+
+    ProxyVector makeProxies(afwTable::SimpleCatalog const & posRefCat) {
+        auto centroidKey = posRefCat.getSchema().find<afwTable::Point<double>>("centroid").key;
+        auto hasCentroidKey = posRefCat.getSchema().find<afwTable::Flag>("hasCentroid").key;
+        ProxyVector r;
+        r.reserve(posRefCat.size());
+        for (afwTable::SimpleCatalog::const_iterator posRefPtr = posRefCat.begin();
+            posRefPtr != posRefCat.end(); ++posRefPtr) {
+            if (!posRefPtr->get(hasCentroidKey)) {
+                throw LSST_EXCEPT(pexExcept::InvalidParameterError, "unknown centroid");
+            }
+            r.push_back(RecordProxy(posRefPtr, posRefPtr->get(centroidKey)));
+        }
+        return r;
+    }
+
+    afwTable::ReferenceMatchVector matchOptimisticB(
+        afwTable::SimpleCatalog const &posRefCat,
+        afwTable::SourceCatalog const &sourceCat,
+        MatchOptimisticBControl const &control,
+        int posRefBegInd,
+        bool verbose
+    ) {
+        control.validate();
+
+        // Select brightest Nsub stars from list of objects
+        // Process both detected from image and external catalog
+        int Nsub = control.numBrightStars;
+        ProxyVector posRefProxyCat = makeProxies(posRefCat);
+        ProxyVector sourceProxyCat = makeProxies(sourceCat);
+        ProxyVector sourceSubCat = selectPoint(
+            sourceProxyCat,
+            sourceCat.getSchema().find<double>(control.sourceFluxField).key,
+            Nsub);
+        ProxyVector posRefSubCat = selectPoint(
+            posRefProxyCat,
+            posRefCat.getSchema().find<double>(control.refFluxField).key,
+            sourceSubCat.size()+25,
+            posRefBegInd);
+        if (verbose) {
+            std::cout << "Catalog sizes: " << sourceSubCat.size() << " " << posRefSubCat.size() << std::endl;
+        }
+
+    /*    
+        CompareProxyFlux cmp = {posRefCat.getSchema().find<double>("flux").key};
+        std::sort(posRefProxyCat.begin(), posRefProxyCat.end(), cmp);
+        cmp = {sourceCat.getTable()->getPsfFluxKey()};
+        std::sort(sourceProxyCat.begin(), sourceProxyCat.end(), cmp);
+    */
```

Comment:
Would prefer to see this deleted or at least disabled with `#if 0` instead of comments.